arm/Microsoft.Network/virtualNetworks/deploy.bicep

```diff
@@ -112,34 +112,71 @@ resource virtualNetwork 'Microsoft.Network/virtualNetworks@2021-05-01' = {
+    networkSecurityGroupName: contains(subnet, 'networkSecurityGroupName') ? subnet.networkSecurityGroupName : ''
+    privateEndpointNetworkPolicies: contains(subnet, 'privateEndpointNetworkPolicies') ? subnet.privateEndpointNetworkPolicies : ''
+    privateLinkServiceNetworkPolicies: contains(subnet, 'privateLinkServiceNetworkPolicies') ? subnet.privateLinkServiceNetworkPolicies : ''
+    routeTableName: contains(subnet, 'routeTableName') ? subnet.routeTableName : ''
+    serviceEndpointPolicies: contains(subnet, 'serviceEndpointPolicies') ? subnet.serviceEndpointPolicies : []
+    serviceEndpoints: contains(subnet, 'serviceEndpoints') ? subnet.serviceEndpoints : []
+  }
+}]
+
+// Local to Remote peering
+module virtualNetwork_peering_local 'virtualNetworkPeerings/deploy.bicep' = [for (peering, index) in virtualNetworkPeerings: {
+  name: '${uniqueString(deployment().name, location)}-virtualNetworkPeering-local-${index}'
   params: {
     localVnetName: name
-    remoteVirtualNetworkId: virtualNetworkPeering.remoteVirtualNetworkId
-    name: contains(virtualNetworkPeering, 'name') ? virtualNetworkPeering.name : '${name}-${last(split(virtualNetworkPeering.remoteVirtualNetworkId, '/'))}'
-    allowForwardedTraffic: contains(virtualNetworkPeering, 'allowForwardedTraffic') ? virtualNetworkPeering.allowForwardedTraffic : true
-    allowGatewayTransit: contains(virtualNetworkPeering, 'allowGatewayTransit') ? virtualNetworkPeering.allowGatewayTransit : false
-    allowVirtualNetworkAccess: contains(virtualNetworkPeering, 'allowVirtualNetworkAccess') ? virtualNetworkPeering.allowVirtualNetworkAccess : true
-    doNotVerifyRemoteGateways: contains(virtualNetworkPeering, 'doNotVerifyRemoteGateways') ? virtualNetworkPeering.doNotVerifyRemoteGateways : true
-    useRemoteGateways: contains(virtualNetworkPeering, 'useRemoteGateways') ? virtualNetworkPeering.useRemoteGateways : false
+    remoteVirtualNetworkId: peering.remoteVirtualNetworkId
+    name: contains(peering, 'name') ? peering.name : '${name}-${last(split(peering.remoteVirtualNetworkId, '/'))}'
+    allowForwardedTraffic: contains(peering, 'allowForwardedTraffic') ? peering.allowForwardedTraffic : true
+    allowGatewayTransit: contains(peering, 'allowGatewayTransit') ? peering.allowGatewayTransit : false
+    allowVirtualNetworkAccess: contains(peering, 'allowVirtualNetworkAccess') ? peering.allowVirtualNetworkAccess : true
+    doNotVerifyRemoteGateways: contains(peering, 'doNotVerifyRemoteGateways') ? peering.doNotVerifyRemoteGateways : true
+    useRemoteGateways: contains(peering, 'useRemoteGateways') ? peering.useRemoteGateways : false
+  }
+  dependsOn: [
+    virtualNetwork_subnets
+  ]
+}]
+
+// Remote to local peering (reverse)
+module virtualNetwork_peering_remote 'virtualNetworkPeerings/deploy.bicep' = [for (peering, index) in virtualNetworkPeerings: if (contains(peering, 'reversePeeringEnabled') ? peering.reversePeeringEnabled == true : false) {
```

Comment:
I think `reverse` can be confusing as it is not part of our original documentation. Maybe the switch `enableRemotePeering` is more aligned? or something around that? Maybe its more of a construct than a module aligned to the doco?

Reply:
Regarding the names I agree. However, regarding the logic itself I'd not for a few reasons
1. I was already asked why the feature isn't there (not the best defense, but still)
2. The 2-way peering is strongly correlated with the rest of the logic
3. It's an optional additional deployment that somebody can enable/disable and still algined to the documentation. The additional parameter I add here are not e.g. on the peering child-resource itself, but the parent. For similar reasons we e.g. need to use enable-flag for the VM extensions.
However, you're correct that it introduces 'additional' logic and hence is up for debate. I'd suggest to either bring it up in the channel or one of the next meetings

Reply:
Updated all references of `reverse` to `remote`.